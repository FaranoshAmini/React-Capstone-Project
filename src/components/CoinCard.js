import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const CoinCard = ({ coin }) => {
  const title = 'Test';

  return (
    <div className="coin-card">
      <div className="coin-img">
        <img src={coin.image} alt="Bitcoin" />
      </div>
      <h3 className="coin-name">{coin.name}</h3>
      <p className="coin-rank">
        <span>Rank: </span>
        <span>{coin.rank}</span>
      </p>
      <p className="coin-price">
        <span>Price: </span>
        <span>
          $
          {coin.price}
        </span>
      </p>
      <div className="details">
        <Link to="/">Details</Link>
      </div>
    </div>
  );
};

CoinCard.propTypes = {
  coin: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    image: PropTypes.string,
    rank: PropTypes.number,
    price: PropTypes.number,
  }).isRequired,
};

export default CoinCard;
