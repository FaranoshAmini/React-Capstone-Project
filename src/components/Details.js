/* eslint-disable react/jsx-one-expression-per-line */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { getAllData } from '../redux/coins/allCoinsReducer';

const Details = () => {
  const { id } = useParams();
  const coinsData = useSelector((state) => state.allCoins);
  const dispatch = useDispatch();
  const coin = coinsData.find((coin) => coin.id === id);
  useEffect(() => {
    if (!coinsData.length) {
      dispatch(getAllData());
    }
  }, []);
  return (
    <section className="coin-detail">
      {coin && (
        <>
          <div className="detail-img">
            <img src={coin.image} alt={coin.name} />
          </div>
          <h3 className="detail-name">{coin.name}</h3>
          <table>
            <tr>
              <th>Symbol</th>
              <td>{coin.symbol}</td>
            </tr>
            <tr>
              <th>Price</th>
              <td>$ {coin.current_price}</td>
            </tr>
            <tr>
              <th>Rank</th>
              <td>{coin.market_cap_rank}</td>
            </tr>
            <tr>
              <th>Total Volume</th>
              <td>{coin.total_volume}</td>
            </tr>
            <tr>
              <th>Circulating Supply</th>
              <td>{coin.circulating_supply}</td>
            </tr>
            <tr>
              <th>Total supply</th>
              <td>{coin.total_supply}</td>
            </tr>
          </table>
        </>
      )}
    </section>
  );
};
export default Details;
