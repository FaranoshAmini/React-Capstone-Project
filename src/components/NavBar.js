import React from 'react';
import { Link } from 'react-router-dom';
import { BiArrowBack } from 'react-icons/bi';

const NavBar = () => (
  <nav>
    <div className="nav-btn">
      <Link to="/" className="btn-link">
        <BiArrowBack />
      </Link>
    </div>
    <div className="logo">
      Crypto
      <span>Coins</span>
    </div>
  </nav>
);

export default NavBar;
