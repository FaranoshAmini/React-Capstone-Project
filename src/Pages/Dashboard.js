import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import CoinCard from '../components/CoinCard';
import { getAllCoins } from '../redux/coins/coinsReducer';

const Dashboard = () => {
  const coinsData = useSelector((state) => state.coins);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!coinsData.length) {
      dispatch(getAllCoins());
    }
  }, []);

  console.log(coinsData);

  return (
    <section className="coins-container">
      {
        coinsData
            && coinsData.map((coin) => <CoinCard key={coin.id} coin={coin} />)
      }
    </section>
  );
};

export default Dashboard;
