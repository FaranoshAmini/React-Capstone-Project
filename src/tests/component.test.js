import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Dashboard from '../Pages/Dashboard';
import CoinCard from '../components/CoinCard';
import NavBar from '../components/NavBar';
import Details from '../components/Details';

describe('UI Tests', () => {
  it('should render Dashboard', () => {
    const tree = render(
      <Provider store={store}>
        <Dashboard />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('contains an image and renders CoinCard component', () => {
    const coins = {
      name: 'Ethereum',
      image: '',
      rank: 2,
      currentPrice: '234',
    };
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <CoinCard coin={coins} />
        </BrowserRouter>
      </Provider>,
    );
    const image = screen.getByRole('img');
    expect(image).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });

  it('should render NavBar', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render Details Component', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
