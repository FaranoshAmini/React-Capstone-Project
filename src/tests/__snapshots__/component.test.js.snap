// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Tests contains an image and renders CoinCard component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="coin-card"
      >
        <div
          class="coin-img"
        >
          <img
            alt="Bitcoin"
            src=""
          />
        </div>
        <h3
          class="coin-name"
        >
          Ethereum
        </h3>
        <p
          class="coin-rank"
        >
          <span>
            Rank: 
          </span>
          <span>
            2
          </span>
        </p>
        <p
          class="coin-price"
        >
          <span>
            Price: 
          </span>
          <span>
            $
          </span>
        </p>
        <div
          class="details"
        >
          <a
            href="/details/undefined"
          >
            Details
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="coin-card"
    >
      <div
        class="coin-img"
      >
        <img
          alt="Bitcoin"
          src=""
        />
      </div>
      <h3
        class="coin-name"
      >
        Ethereum
      </h3>
      <p
        class="coin-rank"
      >
        <span>
          Rank: 
        </span>
        <span>
          2
        </span>
      </p>
      <p
        class="coin-price"
      >
        <span>
          Price: 
        </span>
        <span>
          $
        </span>
      </p>
      <div
        class="details"
      >
        <a
          href="/details/undefined"
        >
          Details
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI Tests should render Dashboard 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="coins-container"
      />
    </div>
  </body>,
  "container": <div>
    <section
      class="coins-container"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI Tests should render Details Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="coin-detail"
      />
    </div>
  </body>,
  "container": <div>
    <section
      class="coin-detail"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI Tests should render NavBar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <div
          class="nav-btn"
        >
          <a
            class="btn-link"
            href="/"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 11H6.414l5.293-5.293-1.414-1.414L2.586 12l7.707 7.707 1.414-1.414L6.414 13H21z"
              />
            </svg>
          </a>
        </div>
        <div
          class="logo"
        >
          Crypto
          <span>
            Coins
          </span>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <div
        class="nav-btn"
      >
        <a
          class="btn-link"
          href="/"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 11H6.414l5.293-5.293-1.414-1.414L2.586 12l7.707 7.707 1.414-1.414L6.414 13H21z"
            />
          </svg>
        </a>
      </div>
      <div
        class="logo"
      >
        Crypto
        <span>
          Coins
        </span>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
